src/Vault.sol

```diff
@@ -493,6 +496,8 @@ contract Vault {
             }
         }
 
+        require(IRegistry(registry).processWithrawal(assetAddresses, assetAmounts), "V_W: Withdrawal failed");
```

Comment:
Would do check before the withdrawals, respect checks effects interactions as much as possible (even if it is here not that important)